go/vt/worker/split_clone.go

```diff
@@ -463,6 +465,35 @@ func (scw *SplitCloneWorker) copy(ctx context.Context) error {
 		}(si.ShardName(), insertChannels[shardIndex])
 	}
 
+	// read the vschema if needed
+	var keyspaceSchema *vindexes.KeyspaceSchema
+	if *useV3ReshardingMode {
+		kschema, err := scw.wr.TopoServer().GetVSchema(ctx, scw.keyspace)
+		if err != nil {
+			return fmt.Errorf("cannot load VSchema for keyspace %v: %v", scw.keyspace, err)
+		}
+
+		var kformal vindexes.KeyspaceFormal
+		if err := json.Unmarshal([]byte(kschema), &kformal); err != nil {
+			return fmt.Errorf("error unmarshalling vschema for keyspace %s: %v", scw.keyspace, err)
+		}
+
+		formal := &vindexes.VSchemaFormal{
+			Keyspaces: map[string]vindexes.KeyspaceFormal{
+				scw.keyspace: kformal,
+			},
+		}
+		vschema, err := vindexes.BuildVSchema(formal)
```

Comment:
I think you already noticed. This code is very similar to vindexes.Validate. We could look at sharing.

I see the rest of the stuff is WIP. I'll comment as things develop.